src/App.jsx

```diff
@@ -137,13 +139,35 @@ function KanbanBoard() {
   const handleTaskUpdate = async (taskId, updates) => {
     if (!selectedTask?.columnId) return;
     await updateTask(selectedTask.columnId, taskId, updates);
+    setSelectedTask(prev => ({
+      ...prev,
+      ...updates
+    }));
   };
 
   const handleTaskDelete = async (taskId) => {
     if (!selectedTask?.columnId) return;
     await deleteTask(selectedTask.columnId, taskId);
   };
 
+  const handleAddTask = async (columnId) => {
+    if (!newTaskContent?.trim()) return;
+    
+    const dueDate = new Date();
+    dueDate.setDate(dueDate.getDate() + 7); // Default to 1 week from now
+    
+    await addTask(columnId, {
+      id: crypto.randomUUID(),
+      content: newTaskContent.trim(),
+      createdAt: dueDate.toISOString(),
+      dueDate: dueDate.toISOString(),
+      labels: []
+    });
+    
+    setNewTaskContent('');
+    setAddingTaskToColumn(null);
```

Comment:
![kody code-review](https://img.shields.io/badge/kody-code--review-brightgreen)

```suggestion
const resetTaskForm = () => {
  setAddingTaskToColumn(null);
  setNewTaskContent('');
};

// Use in handleAddTask and cancel button:
resetTaskForm();
```
Extract the task form reset logic into a separate function to avoid code duplication and improve maintainability



<sub>Talk to Kody by mentioning @kody</sub>

<sub>Was this suggestion helpful? React with 👍 or 👎 to help Kody learn from this interaction.</sub><!-- kody-codereview -->&#8203;
&#8203;